### PR TITLE
Remove offence class for unlisted offences and fix add another

### PR DESCRIPTION
**Pull request by carolinapizatto**

### CODE REVIEW

File: app/views/application_cert_review.html [hunk 1]
```diff
@@ -242,22 +335,20 @@ <h2 class="govuk-heading-m">Interests of Justice</h2>
     </div>
 
 
-    <h1 class="govuk-heading-l">Print application</h1>
+    <!-- <h1 class="govuk-heading-l">Print application</h1> -->
```
Comment (oliverweighell) [inline app/views/application_cert_review.html:338]: Any reason for commenting out rather than deleting?

File: app/views/means.json [hunk 2]
```diff
@@ -0,0 +1,217 @@
+{
```
Comment (oliverweighell) [inline app/views/means.json:1]: Was this file for local testing? Should it be removed from the PR?

File: app/views/x.html [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<div data-module="moj-add-another">
```
Comment (oliverweighell) [inline app/views/x.html:1]: Was this file for local testing? Should it be removed from the PR?

**Review by oliverweighell — CHANGES_REQUESTED** [review]

File: app/views/application_cert_review.html (hunk 1 above)
Comment (carolinapizatto) [reply app/views/application_cert_review.html]: No - I have deleted it now

File: app/routes.js [hunk 4]
```diff
@@ -11,6 +11,7 @@ const applicationsApiUrl = "https://n7ykjge71d.execute-api.eu-west-2.amazonaws.c
 const offencesList = require('./data/offence_list');
 const https = require('https');
 const utils = require('./utils');
+const { constructDate } = require('./utils');
```
[the remark this answers is not in the document]
Comment (arthurashman) [reply app/routes.js]: Probably an auto-add from VSCode I didn't see apologies.

File: app/utils.js [hunk 5]
```diff
@@ -22,6 +23,53 @@ const utils = {
...
   },
+  constructOffences: (req) => {
+    let data = req.session.data
+    let offenceIds = utils.filterOffenceIds(data.offences_search_entry)
+    let offences = offenceIds.map(offenceId => offencesList[offenceId])
+    if (data.offences_manual_entry) {
+      let manual_offences = Array.isArray(data.offences_manual_entry)? data.offences_manual_entry : [data.offences_manual_entry]
+      manual_offences.forEach(offence => offences.push({B: `${offence}`})) 
+    }
+    req.session.data.case_details.offences = offences;
```
[the remark this answers is not in the document]
Comment (arthurashman) [reply app/utils.js:34]: Didn't know this, thank you! Helpful use of lodash too

File: app/utils.js [hunk 6]
```diff
@@ -22,6 +23,53 @@ const utils = {
...
+            offence_date: utils.formatDate(date)
+          });
+        });
+      } else {
+        offencesWithDates.push({
+          offence: data['case_details']['offences'][index].B,
+          offence_class: data['case_details']['offences'][index].D || "not specified",
+          offence_date: utils.formatDate(date)
+        });
+      }
```
[the remark this answers is not in the document]
Comment (arthurashman) [reply app/utils.js]: Thank you! Definitely needed a refactor

**Pull request merged**